src/ds/mod.rs

```diff
@@ -1,122 +1,95 @@
-use crossbeam_channel::{self, Sender, Receiver, unbounded};
 use failure::bail;
 
 use std::thread;
 
-pub(crate) mod state;
 mod conn;
+pub(crate) mod state;
 
-use self::state::*;
 use self::conn::*;
+use self::state::*;
+
+use std::sync::Arc;
+use futures::executor::block_on;
 
-use std::sync::{Arc, Mutex};
-use std::time::Duration;
+use futures_channel::mpsc::{UnboundedSender, unbounded};
 
-use crate::outbound::udp::types::{Request, Alliance};
-use crate::outbound::udp::types::tags::UdpTag;
-use crate::outbound::tcp::*;
-use crate::inbound::tcp::TcpPacket;
-use crate::inbound::udp::types::Trace;
-use crate::Result;
+use crate::proto::tcp::outbound::{GameData, TcpTag};
+use crate::proto::udp::inbound::types::Trace;
+use crate::proto::udp::outbound::types::tags::UdpTag;
+use crate::proto::udp::outbound::types::*;
 use crate::util::ip_from_team_number;
+use crate::{Result, TcpPacket};
 
 /// Represents a connection to the roboRIO acting as a driver station
 ///
 /// This struct will contain relevant functions to update the state of the robot,
 /// and also manages the threads that manage network connections and joysticks
 pub struct DriverStation {
-    thread_tx: Sender<Signal>,
-    thread_rx: Receiver<Signal>,
-    state: Arc<Mutex<State>>,
+    thread_tx: UnboundedSender<Signal>,
     team_number: u32,
+    state: Arc<DsState>,
 }
 
 impl DriverStation {
-
     pub fn new_team(team_number: u32, alliance: Alliance) -> DriverStation {
-        Self::new(&ip_from_team_number(team_number), team_number, alliance)
+        Self::new(&ip_from_team_number(team_number), alliance, team_number)
     }
 
     /// Creates a new driver station for the given alliance station and team number
     /// Connects to the roborio at `ip`. To infer the ip from team_number, use `new_team` instead.
-    pub fn new(ip: &str, team_number: u32, alliance: Alliance) -> DriverStation {
+    pub fn new(ip: &str, alliance: Alliance, team_number: u32) -> DriverStation {
```

Comment:
Any particular reason for changing the order of this?

Reply:
I tore out a lot of stuff initially, the `team_number` value in `DriverStation` being part of that before I restructured the backend to make it possible to change the target without restarting the runtime. Later on, as I was trying to integrate the new library back into my driver station implementation, I found that keeping track of the team number would be valuable and added it back.